extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java

```diff
@@ -21,59 +22,66 @@
 
-        Timer serviceDiscoveryTimer = Timer
+        this.serviceDiscoveryTimer = Timer
                 .builder("stork.service-discovery.duration")
                 .description("The duration of the discovery operation")
-                .tags(tags)
-                .register(registry);
+                .withRegistry(registry);
 
-        Timer serviceSelectionTimer = Timer
+        this.serviceSelectionTimer = Timer
                 .builder("stork.service-selection.duration")
                 .description("The duration of the selection operation ")
-                .tags(tags)
-                .register(registry);
+                .withRegistry(registry);
 
-        Counter serviceDiscoveryFailures = Counter
+        this.serviceDiscoveryFailures = Counter
                 .builder("stork.service-discovery.failures")
-                .description("The number of failures during service discovery").tags(tags)
-                .register(registry);
+                .description("The number of failures during service discovery")
+                .withRegistry(registry);
 
-        Counter serviceSelectionFailures = Counter
+        this.serviceSelectionFailures = Counter
                 .builder("stork.service-selection.failures")
-                .description("The number of failures during service selection.").tags(tags)
-                .register(registry);
+                .description("The number of failures during service selection.")
+                .withRegistry(registry);
+    }
+
+    @Override
+    public StorkObservation create(String serviceName, String serviceDiscoveryType,
+            String serviceSelectionType) {
+        return STORK_METRICS.computeIfAbsent(serviceName + METRICS_SUFFIX,
+                key -> new StorkObservation(serviceName, serviceDiscoveryType, serviceSelectionType,
+                        this));
+    }
+
+    @Override
+    public void complete(StorkObservation observation) {
+        Tags tags = Tags.of(Tag.of("service-name", observation.getServiceName()));
 
-        instanceCounter.increment(observation.getDiscoveredInstancesCount());
-        serviceDiscoveryTimer.record(observation.getServiceDiscoveryDuration().getNano(), TimeUnit.NANOSECONDS);
-        serviceSelectionTimer.record(observation.getServiceSelectionDuration().getNano(), TimeUnit.NANOSECONDS);
+        this.instanceCounter.withTags(tags).increment(observation.getDiscoveredInstancesCount());
+        this.serviceDiscoveryTimer.withTags(tags).record(observation.getServiceDiscoveryDuration().getNano(),
+                TimeUnit.NANOSECONDS);
+        this.serviceSelectionTimer.withTags(tags).record(observation.getServiceSelectionDuration().getNano(),
+                TimeUnit.NANOSECONDS);
 
         if (observation.failure() != null) {
             if (observation.isServiceDiscoverySuccessful()) {
-                serviceSelectionFailures.increment();
+                this.serviceSelectionFailures.withTags(tags).increment();
```

Comment:
There were no tags before... But probably it's better to have them.

Reply:
Yes, I miss the tags in the first commit.

Reply:
When I say before I mean, before the PR... CC @aureamunoz